Increase timeout from 30 seconds to 360 seconds

Reasoning for not upstreaming this patch: Drake-specific option value.

--- mypy/moduleinspect.py
+++ mypy/moduleinspect.py
@@ -157,13 +157,15 @@
 
         Return the value read from the queue, or None if the process unexpectedly died.
         """
-        max_iter = 600
+        timeout_seconds = 720
+        timeout_increment = 0.05
+        max_iter = int(timeout_seconds / timeout_increment)
         n = 0
         while True:
             if n == max_iter:
                 raise RuntimeError("Timeout waiting for subprocess")
             try:
-                return self.results.get(timeout=0.05)
+                return self.results.get(timeout=timeout_increment)
             except queue.Empty:
                 if not self.proc.is_alive():
                     return None
